AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientOperations/AWSMobileClient+Credentials.swift

```diff
@@ -0,0 +1,76 @@
+//
+import Foundation
+
+extension AWSMobileClient {
+    
+    /// Fetches the `AWSCredentials` asynchronously.
+    ///
+    /// - Parameter completionHandler: completionHandler which would have `AWSCredentials` if successfully retrieved, else error.
+    public func getAWSCredentials(_ completionHandler: @escaping(AWSCredentials?, Error?) -> Void) {
+        if self.internalCredentialsProvider == nil {
+            completionHandler(nil, AWSMobileClientError.cognitoIdentityPoolNotConfigured(message: "There is no valid cognito identity pool configured in `awsconfiguration.json`."))
+        }
+
+        let cancellationToken = self.credentialsFetchCancellationSource
+        let operation = AWSAsyncBlockOperation { done in
+            self.internalCredentialsProvider?.credentials(withCancellationToken: cancellationToken).continueWith(block: { (task) -> Any? in
+                // If we have called cancellation already, leave the block without doing anything
+                // with the fetched credentials.
+                if (task.isCancelled || cancellationToken.isCancellationRequested) {
+                    completionHandler(task.result, task.error)
+                    done()
+                    return nil
+                }
+
+                if let error = task.error {
+                    if error._domain == AWSCognitoIdentityErrorDomain
+                        && error._code == AWSCognitoIdentityErrorType.notAuthorized.rawValue
+                        && self.federationProvider == .none {
+
+                        completionHandler(nil, AWSMobileClientError.guestAccessNotAllowed(message: "Your backend is not configured with cognito identity pool to allow guest access. Please allow un-authenticated access to identity pool to use this feature."))
+                        done()
+                    } else if error._domain == AWSCognitoIdentityErrorDomain
+                        && error._code == AWSCognitoIdentityErrorType.notAuthorized.rawValue
+                        && self.federationProvider == .oidcFederation {
+                        // store a reference to the completion handler which we would be calling later on.
+                        self.pendingAWSCredentialsCompletion = { credentials, error in
+                            completionHandler(credentials, error)
+                            done()
+                        }
+
+                        self.mobileClientStatusChanged(
+                            userState: .signedOutFederatedTokensInvalid,
+                            additionalInfo: [
+                                AWSMobileClientConstants.ProviderKey:self.cachedLoginsMap.first!.key
+                            ])
+                    } else {
+                        completionHandler(nil, error)
+                        done()
+                    }
+                } else if let result = task.result {
+                    if(self.federationProvider == .none && self.currentUserState != .guest) {
```

Comment:
nit: formatting

* space after `if`
* drop the parens

Reply:
Done